backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/scheduling/policyunits/CPUPolicyUnit.java

```diff
@@ -47,14 +57,23 @@ public CPUPolicyUnit(PolicyUnit policyUnit,
                 continue;
             }
             Integer cores = SlaValidator.getEffectiveCpuCores(vds, context.getCluster().getCountThreadsAsCores());
-            if (cores != null && (VmCpuCountHelper.isDynamicCpuTopologySet(vm) ?
-                    vm.getCurrentNumOfCpus(false) : vm.getNumOfCpus(false)) > cores) {
-                messages.addMessage(vds.getId(), EngineMessage.VAR__DETAIL__NOT_ENOUGH_CORES.toString());
-                log.debug("Host '{}' has less cores ({}) than vm cores ({})",
-                        vds.getName(),
-                        cores,
-                        VmCpuCountHelper.getDynamicNumOfCpu(vm));
-                continue;
+            // takenCpus are CPUs (threads), we should consider switching it to cores when necessary.
+            takenCpus = context.getCluster().getCountThreadsAsCores() ? takenCpus : (int) Math.ceil(takenCpus / vds.getCpuSockets());
+            int futureCpus = context.getCluster().getCountThreadsAsCores() ? cores - vds.getVmsCoresCount() :
+                    cores - (int) Math.ceil(vds.getVmsCoresCount() / vds.getCpuSockets());
```

Comment:
What does `Math.ceil(vds.getVmsCoresCount() / vds.getCpuSockets())` actually count?

Reply:
Depending on getCountThreadsAsCores. All the takenCpus calculations are per CPUs (threads). If the cluster is set with getCountThreadsAsCores == false, the number we get here is of the cores, not the threads. Here in such case, I also switch the CPUs(threads) to cores.

Reply:
But why dividing by the number of sockets? Shouldn't it be divided by the number of threads?

How does getCountThreadsAsCores() apply to the pinning logic in the VdsCpuUnitHelper? Should it be considered there too?